### PR TITLE
Make a bunch of miscellaneous code tidies

### PR DESCRIPTION
**Pull request by alexwlchan**
I've been gradually building these up locally as I read different files, putting them all in one PR so I can get back to a clean working tree.

### CODE REVIEW

File: catalogue/webapp/components/WorkDetailsSection/WorkDetailsSection.tsx [hunk 1]
```diff
@@ -27,17 +27,13 @@ const SectionWithDivider = styled(Space).attrs({
...
   children,
-}: Props) => {
+}) => {
   const isArchive = useContext(IsArchiveContext);
 
   return (
     <SectionWithDivider isArchive={isArchive}>
       <SpacingSection>
-        {headingText && (
-          <h2 className={`${font('wb', 4)} work-details-heading`}>
```
Comment (alexwlchan) [inline catalogue/webapp/components/WorkDetailsSection/WorkDetailsSection.tsx:37]: This class isn't defined anywhere.

File: content/webapp/components/BookPromo/BookPromo.tsx [hunk 2]
```diff
@@ -101,7 +101,7 @@ const BookPromo: FunctionComponent<Props> = ({ book }) => {
               <LabelsList labels={[{ text: 'Book' }]} />
             </Space>
           </Space>
-          {title && <Title>{title}</Title>}
```
Comment (alexwlchan) [inline content/webapp/components/BookPromo/BookPromo.tsx:104]: `title` is always defined here.

File: content/webapp/components/MediaObjectBase/MediaObjectBase.tsx [hunk 3]
```diff
@@ -143,17 +141,8 @@ const MediaObjectBase: FunctionComponent<Props> = ({
...
-            className={classNames({
-              'spaced-text': true,
-              [font('intr', 5)]: !descriptionIsString,
-            })}
-          >
-            {descriptionIsString ? (
-              <p className={font('intr', 5)}>{description}</p>
-            ) : (
-              description
-            )}
```
Comment (alexwlchan) [inline content/webapp/components/MediaObjectBase/MediaObjectBase.tsx:156]: `descriptionIsString` is a boolean:

* if it's true, we apply `font('intr', 5)` on the `<p>` tag in the first branch of the ternary
* if it's false, we apply `font('intr', 5)` on the `<div>` tag using `classNames`

We always apply the `font('intr', 5)` class; doing it in two places feels like a bit of a code smell – we should be able to apply it once on the `<div>` and be done.

**Review by rcantin-w — APPROVED** [review]
Nice ones!

**Pull request merged**